File: JAVASCRIPT/Cap08/c08ex05.js
// c08ex05.js

const rls = require('readline-sync');

let NOME = new Array(10).fill('').map(() => 
new Array(50).fill('').join(''));
let RESP = 'SIM';
let PESQ = new Array(50).fill('').join('');
let RESPOSTA = new Array(4).fill('').join('');
let I;
let ACHA;

console.log("PESQUISA SEQUENCIAL DE NOMES\n");

for (I = 0; I <= 9; ++I) {
  let DADO = `Entre o ${(I + 1) < 10 ? ` ${I + 1}` : I + 1}o. nome: `
  NOME[I] = rls.question(DADO);
}

while (RESP == 'SIM') {
  PESQ = rls.question('\nEntre o nome a ser pesquisado: ');
  I = 0;
  ACHA = false;

  while (I <= 9 && ACHA == false) {
    if (PESQ == NOME[I]) {
      ACHA = true;
    } else {
      ++I;
    }
  }

  if (ACHA == true) {
    console.log(`\n${PESQ} foi localizado na posicao ${I + 1}`);
  } else {
    console.log(`\n${PESQ} nao foi localizado`);
  }

  do {
    RESPOSTA = rls.question('\nDeseja continuar? (SIM/NAO): ');
    RESP = RESPOSTA.toUpperCase();
    if (RESP != 'SIM' && RESP != 'NAO') {
      console.log('Informe apenas SIM ou NAO.');
    }
  } while (RESP != 'SIM' && RESP != 'NAO');
}

console.log("");
process.stdout.write("Tecle <Enter> para encerrar... ");
rls.question();

File: CSHARP/Cap08/c08ex05.cs
// c08ex05.cs

using System;
using static System.Console;

class Program
{
    static void Main(string[] args)
    {
        string[] NOME = new string[10];

        string RESP = "SIM";
        string PESQ;
        string RESPOSTA;
        int I;
        bool ACHA;

        WriteLine("PESQUISA SEQUENCIAL DE NOMES\n");

        for (I = 0; I <= 9; ++I)
        {
            Write($"Entre o {I + 1,2}o. nome: ");
            NOME[I] = ReadLine();
        }

        while (RESP == "SIM")
        {
            Write("\nEntre o nome a ser pesquisado: ");
            PESQ = ReadLine();
            I = 0;
            ACHA = false;

            while (I <= 9 && ACHA == false)
            {
                if (PESQ == NOME[I])
                    ACHA = true;
                else
                    ++I;
            }

            WriteLine();
            if (ACHA == true)
                WriteLine($"{PESQ} foi localizado na posicao {I + 1}");
            else
                WriteLine($"{PESQ} nao foi localizado");

            do
            {
                Write("\nDeseja continuar? (SIM/NAO): ");
                RESPOSTA = ReadLine();
                RESP = RESPOSTA.ToUpper();
                if (RESP != "SIM" && RESP != "NAO")
                    WriteLine("Informe apenas SIM ou NAO.");
            } while (RESP != "SIM" && RESP != "NAO");
        }

        Write("\nTecle <Enter> para encerrar... ");
        ReadLine();
    }
}
